microbench/src/main/java/io/netty/buffer/ByteBufAccessBenchmark.java

```diff
@@ -0,0 +1,147 @@
+/*
+* Copyright 2019 The Netty Project
+*
+* The Netty Project licenses this file to you under the Apache License,
+* version 2.0 (the "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at:
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*/
+package io.netty.buffer;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.internal.PlatformDependent;
+
+@Warmup(iterations = 5, time = 1500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 1500, timeUnit = TimeUnit.MILLISECONDS)
+@Fork(3)
+public class ByteBufAccessBenchmark extends AbstractMicrobenchmark {
```

Comment:
Using Sampling or Average instead of Ops/s produces _much_ more clear results and allows you to do things like add two benchmarks together to get a rough idea of how long a composite operation may take (and then you can also compare that against a benchmark).

Reply:
@njhill @ejona86 
Beware the precision & granularity of `System::nanoTime` used to collect the averageTime as shown on https://shipilev.net/blog/2014/nanotrusting-nanotime/.
The risk is that operations with < 15 ns cost won't be correctly measured in many *nix envs!

Reply:
@franz1981, I don't know what you're talking about. Average is just "total amount of time / number of runs". Aleksey even shows the code explicitly in latency_nanotime_AverageTime_measurementLoop and notes it is okay. I would agree sampling could have trouble, although the aliasing should also be obvious within the results.

Reply:
@ejona86 I'm saying that If you have a precision on nanoTime of 15 ns you wouldn't be able to get meaningful averages for operations that take less then this time to be executed...